### PR TITLE
Order-Order Details and CRUD Repository V1.0

### PR DESCRIPTION
**Pull request by jamendez216**

### CODE REVIEW

File: odyssey-api/CRUD/CRUDModel.cs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+            dbSet = _context.Set<T>();
+        }
+
+        #endregion
+
+
+        #region CRUD
+
+        #region Select
+        public IEnumerable<T> ObtenerTodos()
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Espanglish (?

File: odyssey-api/CRUD/CRUDModel.cs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+            {
+                return dbSet.ToList();
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        public T Obtener(int id)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Espanglish

File: odyssey-api/CRUD/CRUDModel.cs [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+                return dbSet.Find(id);
+            }        
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+
+        public T Obtener(string id)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Espanglish too lmao

File: odyssey-api/CRUD/CRUDModel.cs [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+            {
+                throw e;
+            }
+        }
+
+        #endregion
+
+
+        #region Insert
+        public bool Insertar(T entity)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Espanglish

File: odyssey-api/CRUD/CRUDModel.cs [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+                }
+                catch (Exception e)
+                {
+                    tran.Rollback();
+                    throw e.InnerException ?? e;
+                }
+
+            }
+        }
+        public bool Insertar(List<T> entity)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Espanglish

File: odyssey-api/CRUD/CRUDModel.cs [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+                throw e;
+            }
+        }
+
+        #endregion
+
+
+        #region Update
+
+        public bool Actualizaaar(T nuevo)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Dude wtf

File: odyssey-api/CRUD/CRUDModel.cs [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+                dataContext.Entry(nuevo).State = EntityState.Modified;
+                return dataContext.SaveChanges() > 0;
+            }
+            catch (Exception e)
+            {
+                throw e.InnerException ?? e;
+            }
+        }
+
+        public bool Actualizar(T nuevo)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Whats the difference between this and Actualizaaar?

File: odyssey-api/CRUD/CRUDModel.cs [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+        {
+            try
+            {
+                dbSet.Update(nuevo);
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Spanish param

File: odyssey-api/CRUD/CRUDModel.cs [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+                return dataContext.SaveChanges() > 0;
+            }
+            catch (Exception e)
+            {
+                throw e.InnerException ?? e;
+            }
+        }
+        public bool Actualizar(IEnumerable<T> nuevo)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Spanish

File: odyssey-api/CRUD/CRUDModel.cs [hunk 10]
```diff
@@ -0,0 +1,346 @@
...
+                }
+                return dataContext.SaveChanges() > 0;
+            }
+            catch (Exception e)
+            {
+                throw e.InnerException ?? e;
+            }
+        }
+
+        public bool Actualizaaar(IEnumerable<T> nuevo)
```
Comment (dfeliz) [inline odyssey-api/CRUD/CRUDModel.cs]: Spanglish

File: odyssey-api/CRUD/ICRUDModel.cs [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public interface ICRUDModel<T> where T : class
+    {
+        IEnumerable<T> Get();
+        T Get(int id);
+        T Get(string id);
+        bool Insert(T entity);
+        bool Insert(List<T> entity);
+        T AddGetObject(T entity);
+        bool Update(T nuevo);
```
Comment (dfeliz) [inline odyssey-api/CRUD/ICRUDModel.cs]: *breathes*

**Comment by angelmadames** [comment]
We'll merge this and improve the code as it goes.

**Pull request merged**